From e4851286ec620bfe7ed18670f4de4393dfff4ffc Mon Sep 17 00:00:00 2001
From: Juha Kallioinen <juha.kallioinen@jolla.com>
Date: Thu, 16 Jan 2014 16:58:52 +0000
Subject: [PATCH] Implement macro to skip install processing step

if %_skip_install_processing is set, the %install script will not
process any of the installed files. This is used in Mer SDK to
speed up deployment when an actual RPM package is not built.

Signed-off-by: Juha Kallioinen <juha.kallioinen@jolla.com>
---
 build/files.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/build/files.c b/build/files.c
index 9fa9e29..7763232 100644
--- a/build/files.c
+++ b/build/files.c
@@ -2360,6 +2360,12 @@ rpmRC processBinaryFiles(rpmSpec spec, rpmBuildPkgFlags pkgFlags,
     Package pkg;
     rpmRC rc = RPMRC_OK;
     char *buildroot;
+    int stop_processing = rpmExpandNumeric("%{?_skip_install_processing}");
+
+    if (stop_processing) {
+      rpmlog(RPMLOG_NOTICE, _("Skip processing files\n"));
+      return rc;
+    }
     
     check_fileList = newStringBuf();
     buildroot = rpmGenPath(spec->rootDir, spec->buildRoot, NULL);
-- 
1.8.3-rc3

